tests/manage/z_cluster/nodes/test_toleration.py

```diff
@@ -0,0 +1,167 @@
+
+
+def check_taint_on_nodes(taint=None):
+    """
+    Function to check for particular taint on nodes
+
+    Args:
+        taint (str): The taint to check on nodes
+
+    """
+    taint = taint if taint else "node.ocs.openshift.io/storage=true:NoSchedule"
+    obj = OCP()
+    ocs_nodes = get_labeled_nodes(label="cluster.ocs.openshift.io/openshift-storage=")
+    flag = 0
+    for node in ocs_nodes:
+        output = obj.exec_oc_cmd(
+            command=f"describe node {node}", out_yaml_format=False
+        ).split("\n")
+        for line in output:
+            if "Taints:" in line:
+                taint_out = re.search(taint, line)
+                if taint_out:
+                    logger.info(f"{node} has taint {taint}")
+                    flag = 1
+                    continue
+    if flag:
+        return bool(flag)
+
+
+def taint_nodes(nodes_to_taint=None):
+    """
+    Function to taint nodes
+
+    Args:
+        nodes_to_taint (list): Nodes to taint
+
+    """
+    if not check_taint_on_nodes():
+        ocp = OCP()
+        ocs_nodes = get_labeled_nodes(
+            label="cluster.ocs.openshift.io/openshift-storage="
+        )
+        nodes_to_taint = nodes_to_taint if nodes_to_taint else ocs_nodes
+        logger.info(
+            f"Taint nodes: {nodes_to_taint} with taint: "
+            f"{constants.OPERATOR_NODE_TAINT}"
+        )
+        for node in nodes_to_taint:
+            taint_cmd = f"adm taint nodes {node} {constants.OPERATOR_NODE_TAINT}"
+            ocp.exec_oc_cmd(command=taint_cmd)
+    else:
+        logger.info(f"Nodes already have taint {constants.OPERATOR_NODE_TAINT} ")
+
+
+def untaint_nodes(taint=None, nodes_to_taint=None):
+    """
+    Function to remove taints from nodes
+
+    Args:
+        taint (str): taint to use
+        nodes_to_taint (list): list of nodes to untaint
+
+    """
+    if check_taint_on_nodes():
+        ocp = OCP()
+        ocs_nodes = get_labeled_nodes(
+            label="cluster.ocs.openshift.io/openshift-storage="
+        )
+        taint = taint if taint else "node.ocs.openshift.io/storage=true:NoSchedule"
+        nodes_to_taint = nodes_to_taint if nodes_to_taint else ocs_nodes
+        for node in nodes_to_taint:
+            taint_cmd = f"adm taint nodes {node} {taint}-"
+            ocp.exec_oc_cmd(command=taint_cmd)
+            logger.info(f"Untainted {node}")
+
+
+@tier4
+@tier4a
+class TestNodesRestart(E2ETest):
+    """
+    1. Check if nodes are tainted
+    2. Taint ocs nodes if not tainted
+    3. Check for tolerations on all pod
+    4. Respin all ocs pods and check if it runs on ocs nodes
+    5. Untaint nodes
+
+    """
+
+    @pytest.fixture()
+    def teardown(self, request, nodes):
+        """
+        Make sure all nodes are up again
+
+        """
+
+        def finalizer():
+            untaint_nodes()
+
+        request.addfinalizer(finalizer)
+
+    # @pytest.mark.parametrize(
```

Comment:
You will want to use the parametrize to urn taint_nodes or untain_nodes with different ploarion IDs right?
As of now I see it's just commented out here and without any another action different for these two cases. So I guess this is in TODO and you will do it soon.